app/test/unit/stores/updates/update-remote-url-test.ts

```diff
@@ -0,0 +1,88 @@
+import { GitStore, RepositoriesStore } from '../../../../src/lib/stores'
+import { TestRepositoriesDatabase } from '../../../helpers/databases'
+import { IAPIRepository } from '../../../../src/lib/api'
+import { updateRemoteUrl } from '../../../../src/lib/stores/updates/update-remote-url'
+import { shell } from '../../../helpers/test-app-shell'
+import { setupFixtureRepository } from '../../../helpers/repositories'
+import { addRemote } from '../../../../src/lib/git'
+
+describe('Update remote url', () => {
+  const apiRepository: IAPIRepository = {
+    clone_url: 'https://github.com/my-user/my-repo',
+    ssh_url: 'git@github.com:my-user/my-repo.git',
+    html_url: 'https://github.com/my-user/my-repo',
+    name: 'my-repo',
+    owner: {
+      id: 42,
+      url: 'https://github.com/my-user',
+      login: 'my-user',
+      avatar_url: 'https://github.com/my-user.png',
+      type: 'User',
+    },
+    private: true,
+    fork: false,
+    default_branch: 'master',
+    parent: null,
+    pushed_at: '1995-12-17T03:24:00',
+  }
+
+  let gitStore: GitStore
+
+  const createRepository = async (remoteUrl = apiRepository.clone_url) => {
+    const db = new TestRepositoriesDatabase()
+    await db.reset()
+    const repositoriesStore = new RepositoriesStore(db)
+
+    const repoPath = await setupFixtureRepository('test-repo')
+    const blankRepo = await repositoriesStore.addRepository(repoPath)
+    const repository = await repositoriesStore.updateGitHubRepository(
+      blankRepo,
+      '',
+      apiRepository
+    )
+    await addRemote(repository, 'origin', remoteUrl)
+    gitStore = new GitStore(repository, shell)
+    await gitStore.loadRemotes()
+
+    return { repository, gitStore }
```

Comment:
If there is an easier way to get the repo and gitstore setup for the test I'd love to hear about it!

Reply:
i don't think i know of one 😿

this looks like a more complex case since you're dealing with github remotes